client/src/driver/BitBakeProjectScanner.ts

```diff
@@ -397,41 +403,81 @@ You should adjust your docker volumes to use the same URIs as those present on y
   }
 
   private async scanForRecipesPath (): Promise<void> {
-    const tmpFiles = this.searchFiles(this._recipesFileExtension)
+    const showRecipeFileNameOutput = await this.executeBitBakeCommand('bitbake-layers show-recipes -f')
+    if (showRecipeFileNameOutput.status !== 0) {
+      logger.error('Failed to scan recipes path')
+      throw new Error('Failed to scan recipes path')
+    }
 
-    for (const file of tmpFiles) {
-      const recipeName: string = file.name.split(/[_]/g)[0]
+    const output = showRecipeFileNameOutput.output.toString()
+    const splittedOutput = output.split(/\r?\n/g)
+    const startingIndex = splittedOutput.findIndex((line) => line.includes('Available recipes'))
 
-      const element: ElementInfo | undefined = this._bitbakeScanResult._recipes.find((obj: ElementInfo): boolean => {
-        return obj.name === recipeName
-      })
+    const allFileNames: string[] = []
+    if (startingIndex === -1) {
+      logger.error('[scanForRecipesPath] Failed to find available recipes')
+      return
+    }
+
+    const recipePathRegex = /(.*\.bb)/
 
-      if (element !== undefined) {
-        element.path = file.path
+    // All lines after the '=== Available recipes: ===' line are file names, and only keep the valid ones
+    allFileNames.push(...splittedOutput.slice(startingIndex + 1).filter((line) => !line.includes('(skipped') && recipePathRegex.exec(line) !== null))
+
+    const notMatchedFileNames: string[] = []
+    for (const fileName of allFileNames) {
+      const match = recipePathRegex.exec(fileName)
+      if (match === null) {
+        continue
       }
-    }
 
-    if (this._shouldDeepExamine) {
-      const recipesWithOutPath: ElementInfo[] = this._bitbakeScanResult._recipes.filter((obj: ElementInfo): boolean => {
-        return obj.path === undefined
+      const recipePath = match[0].trim()
+      const foundRecipe = this._bitbakeScanResult._recipes.find((recipe) => {
+        return recipe.name === extractRecipeName(recipePath)
       })
+      if (foundRecipe !== undefined) {
+        foundRecipe.path = path.parse(recipePath)
```

Comment:
You have to call resolveContainerPath on all paths reported by bitbake commands to be compatible with kas. I got some recipes still pointing to the /work mount point binding.

Reply:
Added for the one produced by the command, please re-verify if anywhere else requires it.

Reply:
Some are still missing (should not report /work here)
![image](https://github.com/yoctoproject/vscode-bitbake/assets/19970889/ceacf1e9-bcb9-4ff7-b21f-d208a05d34ba)

Reply:
Pushed a commit to fix this on this branch

Reply:
Pushed a commit with the missing ones.